### DISCUSSION TITLE
Invalidate Apollo cache after mutation *or* vanilla HttpClient?

### DISCUSSION THREAD

**Discussion by mattdsteele** [the-guild-org/apollo-angular#1532] [General]
Similar conversations, but I wasn't able to find a specific answer here: https://github.com/kamilkisiela/apollo-angular/issues/1486 and https://github.com/apollographql/apollo-feature-requests/issues/5

Our use case: our data updates are split between Apollo mutations, and POST calls using Angular's vanilla HttpClient. In both scenarios, an update could require us to invalidate the query cache.

I can't find a good way to invalidate the Apollo cache, particularly if it's happening outside Apollo's context.

We wouldn't have to get too fancy or surgical with the cache invalidation; even something that invalidates everything in the Apollo cache would be sufficient (our updates are infrequent)

Any suggestions?

**Comment by kamilkisiela** [comment]
There's a new API for cache in Apollo Angular 2.0 and Apollo Client 3.0, take a look at ["Garbage collection and cache eviction"](https://apollo-angular.com/docs/caching/garbage-collection). Maybe something interesting for you.
